cli/cook/subcommands/kill.py

```diff
@@ -0,0 +1,109 @@
+from cook import http, colors
+from cook.querying import query, print_no_data
+from cook.util import strip_all, print_info
+
+
+def guard_against_duplicates(uuids, query_result):
+    """
+    Checks for UUIDs that are duplicated and throws if any are found (unlikely to happen in the wild, but it could)
+    """
+    if query_result['count'] == 1:
+        return
+
+    uuid_to_entries = {}
+    duplicate_uuids = []
+
+    def add(uuid, entity_type, cluster):
+        entry_map = {'type': entity_type, 'cluster_name': cluster}
+        if uuid in uuid_to_entries:
+            uuid_to_entries[uuid].append(entry_map)
+            duplicate_uuids.append(uuid)
+        else:
+            uuid_to_entries[uuid] = [entry_map]
+
+    for cluster_name, entities in query_result['clusters'].items():
+        jobs = entities['jobs']
+        for job in jobs:
+            add(job['uuid'], 'job', cluster_name)
+
+        instances = [i for j in entities['instances'] for i in j['instances'] if i['task_id'] in uuids]
```

Comment:
This looks like a typo: `for j in entities['instances']`
... but apparently it isn't? It would be nice if `entities['instances']` was actually a sequence of instances, and there could be a more clearly named member for the instance-jobs sequence being used here.

Also, I'd prefer having this expanded into an imperative nested for-loop.

This is irrelevant if you expand this into nested loops, but on a site note: I think it's usually more efficient to use a generator than a list comprehension if you don't actually need a list (i.e., swapping the `[ ... ]` with `( ... )` should result in slightly faster code).

Reply:
I also prefer the nested loops to the list comprehension here.

Reply:
:+1: I moved this code to the `query_entities` function, because it was being used by other subcommands too. I also changed it to loops.